### PR TITLE
Refactor java package name from cvc5 to io.github.cvc5.api

### PR DESCRIPTION
**Pull request by mudathirmahgoub**
This PR refactors java package name from cvc5 to io.github.cvc5.api.
It also refactor the names of cpp and java files.

### CODE REVIEW

File: src/api/java/CMakeLists.txt [hunk 1]
```diff
@@ -14,15 +14,15 @@
...
+file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/src/api/java")
 set(JNI_DIR "${CMAKE_CURRENT_BINARY_DIR}/src/api/java/jni")
 file(MAKE_DIRECTORY ${JNI_DIR})
 
 # Generate cvc5/Kind.java
 configure_file(genkinds.py.in genkinds.py)
 
 set(JAVA_KIND_FILE
-  "${CMAKE_CURRENT_BINARY_DIR}/src/api/java/cvc5/Kind.java"
+  "${CMAKE_CURRENT_BINARY_DIR}/src/api/java/Kind.java"
```
Comment (nafur) [inline src/api/java/CMakeLists.txt:25]: Shouldn't this also go to `src/api/java/io/github/cvc5/api/`?

File: src/api/java/jni/java_api.h [hunk 2]
```diff
@@ -13,31 +13,40 @@
  * The cvc5 Java API.
  */
 
-#include <jni.h>
-
 #ifndef CVC5__JAVA_API_H
 #define CVC5__JAVA_API_H
 
+#include <jni.h>
```
Comment (nafur) [inline src/api/java/jni/java_api.h:19]: The filename suggests that this file contains the main api, while it only holds a few helpers. Maybe rename to `api_utilities`?

File: test/unit/api/java/ResultTest.java [hunk 3]
```diff
@@ -13,10 +13,9 @@
  * Black box testing of the Result class
  */
 
-package cvc5;
```
Comment (nafur) [inline test/unit/api/java/ResultTest.java:16]: shouldn't it be in the `tests` package?

File: src/api/java/CMakeLists.txt (hunk 1 above)
Comment (mudathirmahgoub) [reply src/api/java/CMakeLists.txt]: Yes. Done

File: src/api/java/jni/java_api.h (hunk 2 above)
Comment (mudathirmahgoub) [reply src/api/java/jni/java_api.h]: Renamed to `api_utilities`

File: test/unit/api/java/ResultTest.java (hunk 3 above)
Comment (mudathirmahgoub) [reply test/unit/api/java/ResultTest.java:16]: Yes. I missed it here. Thanks

**Pull request merged**